01_fib/jonathanheaden/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func fib(n int) {
+	a, b := 0, 1
+	var prefix string
+	var neg = false
+	if n < 0 {
+		n *= -1
+		neg = true
+	}
+	for i := 0; i <= n; i++ {
+		if prefix = ""; neg && (i%2 == 0) {
+			prefix = "-"
+		}
+		switch i {
```

Comment:
No need to have a switch on loop index inside look - might as well do it before imo.